common/workunit/workunit.hpp

```diff
@@ -1190,6 +1190,18 @@ interface IConstWorkUnitInfo : extends IInterface
     virtual IConstWUAppValueIterator & getApplicationValues() const = 0;
 };
 
+interface IConstWUProcessLogInfo : extends IInterface
+{
+    virtual const char *getLogSpec() const = 0;
+    virtual const char *getProcessName() const = 0;
+    virtual unsigned getProcessID() const = 0;
+    virtual const char *getGroupName() const = 0;
+    virtual const char *getLogDate() const = 0;
+    virtual const char *getPattern() const = 0;
```

Comment:
difficult to say without seeing how this new interface would be used, but feels like some of these should internal implementation details, e.g. how does the caller use groupname (especially if there is no group name because the WU was imported).

Feels like the caller should be asking for the log, and the implementtion should be using these details to get it, so the caller doesn't have to concern itself with the internal details of how to access it.